src/System.Private.CoreLib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -80,6 +80,15 @@ public Dictionary(int capacity, IEqualityComparer<TKey> comparer)
                 // To start, move off default comparer for string which is randomised
                 _comparer = (IEqualityComparer<TKey>)NonRandomizedStringEqualityComparer.Default;
             }
+
+            // Default initialization to avoid null check for every insert
+            // Reads will fail and first add cause a resize into a real array
+            if (_entries is null)
```

Comment:
Won't it always be null?

Reply:
If capacity passed is > 0 we call Initialize(capacity)

Reply:
Then wouldn't this be an else off of that?  Or initialize these first and then do that check?

Reply:
Yep